portfolio/src/main/webapp/script.js

```diff
@@ -98,8 +113,22 @@ function createListComment(comment) {
   return liElemName;
 }
 
-function createErrorMssg(errorJson) {
+function createErrorMssg(errorMssg) {
   errMssg = document.createElement('p');
-  errMssg.innerText = errorJson;
+  errMssg.innerText = errorMssg;
   return errMssg;
 }
+
+function addLogInOutButton(url, isLoggedIn) {
+  const link = document.getElementById('login');
```

Comment:
nit: This is a real nit pick and totally up to you, but it might be nicer to use a div and populate it all at once. For example your HTML could have
```html
<div id="login-container"></div>
```
with Javascript that creates and adds the button:
```javascript
const loginLink = document.createElement('a');
...
const button = document.createElement('button');
loginLink.appendChild(button);
...
document.getElementById('login-button').appendChild(loginLink);
```